### PR TITLE
121 chronokvs basic implementation

### PR DESCRIPTION
**Pull request by iameneko**
Most basic implementation of the ChronoKVS. Just to work as a baseline for future development. 

[Documentation](https://docs.google.com/presentation/d/1HZiyFmss9aiAgvWVG2Oplr-zffrw-KaTf5ZOMrYCV80/edit#slide=id.p)

### CODE REVIEW

File: Plugins/chronokvs/examples/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.17)
```
Comment (fkengun) [inline Plugins/chronokvs/examples/CMakeLists.txt]: better to match the CMake version in other directory, which is 3.19.

File: Plugins/chronokvs/src/chronokvs_memorymanager.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#define CHRONOKVS_CHRONOKVS_MEMORYMANAGER_H
+
+#include <string>
+#include <vector>
+#include <utility>
+#include <unordered_map>
+
+namespace chronokvs
+{
+class MemoryManager
```
Comment (fkengun) [inline Plugins/chronokvs/src/chronokvs_memorymanager.h]: my understanding of this naming is for memorization of the mapping of key->vector<timestamp>. but I am expecting a memory allocation manager by its name. I think a straightforward `KeyToTimestampMappingManager` or something similar is better.

File: Plugins/chronokvs/src/chronokvs_mapper.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    std::vector <std::string> serializedEvents = chronoClient->retrieveEvents(timestamp);
+    std::vector <std::string> values;
+
+    for(const std::string &serializedEvent: serializedEvents)
+    {
+        auto keyValue = deserialize(serializedEvent);
+        if(keyValue.first == key)
+        {
+            // Add the value to values vector
+            values.push_back(keyValue.second);
```
Comment (fkengun) [inline Plugins/chronokvs/src/chronokvs_mapper.cpp]: is it possible to break here to save some time? a timestamp may associate with multiple events. but only one of them should match the provided key. is that correct?

**Review by fkengun — COMMENTED** [review]
minor comments are left.

File: Plugins/chronokvs/src/chronokvs_mapper.cpp (hunk 3 above)
Comment (iameneko) [reply Plugins/chronokvs/src/chronokvs_mapper.cpp]: Agree. Not only that, but this function should return a single value (string) instead of a vector

**Review by fkengun — APPROVED** [review]

**Pull request merged**